curriculum/challenges/english/25-front-end-development/lab-depth-first-search/587d825d367417b2b2512c96.md

```diff
@@ -0,0 +1,212 @@
+---
+id: 587d825d367417b2b2512c96
+title: Implement the Depth-First Search Algorithm
+challengeType: 14
+dashedName: implement-the-depth-first-search-algorithm
+---
+
+# --description--
+
+<dfn>Depth-first search</dfn> (DFS) is an algorithm used to search through data structures such as trees or graphs. Starting from a root node, the search first goes down a path of edges as far as it can. Once it reaches one end of a path, the search will backtrack to the last node with an un-visited edge path and continue searching down the un-visited edge path.
```

Comment:
```suggestion
<dfn>Depth-first search</dfn> (DFS) is an algorithm used to search through data structures such as trees or graphs. Starting from a root node, the search first goes down a path of edges as far as it can. Once it reaches one end of a path, the search will backtrack to the last node with an unvisited edge path and continue searching down the unvisited edge path.
```